### PR TITLE
Added mid and exit flow

### PR DESCRIPTION
**Pull request by iamamanagrawal**

### CODE REVIEW

File: Xception/xception.py [hunk 1]
```diff
@@ -177,3 +177,188 @@ def forward(self,x):
...
+        super(MiddleflowSeperable,self).__init__()
+
+        # 1st branch
+        self.sepconv1 = DepthwiseSeparable(in_channel=in_channel,out_channel=out_channel,kernel_size=3)
+        self.sepconv2 = DepthwiseSeparable(in_channel=out_channel,out_channel=out_channel,kernel_size=3)
+        self.sepconv3 = DepthwiseSeparable(in_channel=out_channel,out_channel=out_channel,kernel_size=3)
+        self.relu = nn.ReLU()
+
+        # 2nd branch
+        self.conv = nn.Conv2d(in_channels=in_channel,out_channels=out_channel,kernel_size=1,stride=1)
```
Comment (mabhay3420) [inline Xception/xception.py]: The left branch do not involve a convolution layer so y = x .

File: Xception/xception.py [hunk 2]
```diff
@@ -177,3 +177,188 @@ def forward(self,x):
...
+    This part contains depthwise separable convolutions and is repeated 3 times in original implementation.
+
+        in_channel, out_channel : Both of them are actually equal!
+        kernel_size = 3 : For all repetitions
+    """
+    def __init__(self,in_channel,out_channel):
+        super(MiddleflowSeperable,self).__init__()
+
+        # 1st branch
+        self.sepconv1 = DepthwiseSeparable(in_channel=in_channel,out_channel=out_channel,kernel_size=3)
```
Comment (mabhay3420) [inline Xception/xception.py]: Change names to sep1,sep2,... so as to be consistent.

File: Xception/xception.py [hunk 3]
```diff
@@ -177,3 +177,188 @@ def forward(self,x):
...
+        MiddleFlowSeperable is repeated 8 times 
+    
+    input_size = (19,19,728)
+    output_size = (19,19,728)       
+    """
+    def __init__(self):
+        super(MiddleFlow,self).__init__()
+        self.sep = MiddleflowSeperable(in_channel=728,out_channel=728)
+    def forward(self,x):
+        for i in range(0,8):
```
Comment (mabhay3420) [inline Xception/xception.py]: range(8) will work.

File: Xception/xception.py [hunk 4]
```diff
@@ -177,3 +177,188 @@ def forward(self,x):
...
+
+
+# In[17]:
+
+
+class ExitFlow(nn.Module):
+    """
+    This part contains ExitFlowSeperable part with 2 different depthwise seperable convolutions followed by Global Avgerage Pool(Avg Pool of kernel size 10) and connecting with output layer
+
+    input_size  :(19,19,728)
```
Comment (mabhay3420) [inline Xception/xception.py]: (728,19,19) or 19 x 19 x 728

**Review by mabhay3420 — CHANGES_REQUESTED** [review]

File: Xception/xception.py (hunk 4 above)
Comment (iamamanagrawal) [reply Xception/xception.py]: (728,19,19) is equivalent to 19x19x728 in pytorch

File: Xception/xception.py (hunk 3 above)
Comment (iamamanagrawal) [reply Xception/xception.py]: Both are same

**Comment by mabhay3420** [comment]
"Note that all Convolution and SeparableConvolution layers are followed by batch normalization [7] (not
included in the diagram)." : Need to Update whole implementation.

File: Xception/xception.py (hunk 4 above)
Comment (mabhay3420) [reply Xception/xception.py]: It will be better Using (728,19,19) in place of (19,19,728) If size is written in tuple form,

**Comment by iamamanagrawal** [comment]
> "Note that all Convolution and SeparableConvolution layers are followed by batch normalization [7] (not
> included in the diagram)." : Need to Update whole implementation.

Ok I will do that

File: Xception/xception.py [hunk 5]
```diff
@@ -158,15 +166,23 @@ class EntryFlow(nn.Module):
...
 
     def forward(self,x):
         x = self.conv(x)
+        x = self.bnm_conv(x)
         x = self.sep1(x)
+        x = self.bnm1(x)
         x = self.sep2(x)
+        x = self.bnm2(x)
         x = self.sep3(x)
+        x = self.bnm3(x)
```
Comment (mabhay3420) [inline Xception/xception.py:186]: No need of batch normalization here. Already updated both conv and sep layers.

File: Xception/xception.py [hunk 6]
```diff
@@ -231,15 +251,16 @@ class MiddleFlow(nn.Module):
...
+    output_size = (728,19,19)       
     """
     def __init__(self):
         super(MiddleFlow,self).__init__()
         self.sep = MiddleflowSeperable(in_channel=728,out_channel=728)
+        self.bnm = nn.BatchNorm2d(728)
     def forward(self,x):
         for i in range(8):
-            x = self.sep(x)
+            x = self.bnm(self.sep(x))
```
Comment (mabhay3420) [inline Xception/xception.py]: No need for bnm here.

**Review by mabhay3420 — APPROVED** [review]
Will add a final pr after merging this : remove bnms and update Readme
